podcasts/NowPlayingPlayerItemViewController+Shelf.swift

```diff
@@ -333,16 +346,39 @@ extension NowPlayingPlayerItemViewController: NowPlayingActionsDelegate {
 
     private func showSleepPanel() {
         let sleepController = SleepTimerViewController()
-        let bottomSheet = MDCBottomSheetController(contentViewController: sleepController)
-        roundCorners(bottomSheet: bottomSheet)
-        present(bottomSheet, animated: true, completion: nil)
+
+        if let sheetController = sleepController.sheetPresentationController {
+            if #available(iOS 16.0, *) {
+                let hostingControllerHeight = view.bounds.height
+                let sheetDetentHeight = hostingControllerHeight * 0.6
+                sheetController.detents = [.custom(resolver: { _ in sheetDetentHeight })]
+            } else {
+                sheetController.detents = [.medium()]
+            }
+            // The Sleep Timer VC implements its own grabber UI.
+            sheetController.prefersGrabberVisible = false
+        }
+
+        present(sleepController, animated: true, completion: nil)
     }
 
     private func showEffectsPanel() {
         let effectsController = EffectsViewController()
-        let bottomSheet = MDCBottomSheetController(contentViewController: effectsController)
-        roundCorners(bottomSheet: bottomSheet)
-        present(bottomSheet, animated: true, completion: nil)
+
+        if let sheetController = effectsController.sheetPresentationController {
+            if #available(iOS 16.0, *) {
+                let hostingControllerHeight = view.bounds.height
+                let sheetDetentHeight = hostingControllerHeight * 0.5
+                sheetController.detents = [.custom(resolver: { _ in sheetDetentHeight })]
+            } else {
+                sheetController.detents = [.medium()]
```

Comment:
<img src="https://github.com/Automattic/pocket-casts-ios/assets/651601/2547c947-194b-446c-be8d-88dbc97155ca" width=320>

As you can see from the screenshot above  for smaller devices, like the iPhone SE, we need to adapt the height to fit all the content.

This happens because unlike the MDC wrapper that sizes the bottomsheet using the VC preferred  size the system UI uses a static size that does not adapt to the size.


We can do the following for smaller devices (width is smaller than 375pt)
- iOS 16 and above: increase the height ratio to `0.7`
- IOS 15: use the `[.large]` detent

Reply:
Another issue that we can see in iOS 15 is that the Effects does not fit the full width of the BottomSheet.

<img src="https://github.com/Automattic/pocket-casts-ios/assets/651601/dbd4d384-7435-4e99-85f0-f76796f3cefa" width=320>

This again was that MDC adapted the width of the bottom sheet to the preferred size of the VC, but the system BottomSheet doesn't.

This can be fixed by adding the following line on `viewDidLoad` method  of EffectsViewController:

`view.translatesAutoresizingMaskIntoConstraints = false`